File: ocr_reader.js
// ocr_reader.js — JavaScript версия

const Tesseract = require('tesseract.js');
const fs = require('fs');
const path = require('path');

async function main() {
    const args = process.argv.slice(2);
    let imagePath = null;
    let lang = 'eng';
    let outputFile = null;

    for (let i = 0; i < args.length; i++) {
        if (args[i] === '-l' || args[i] === '--lang') {
            lang = args[++i];
        } else if (args[i] === '-o' || args[i] === '--output') {
            outputFile = args[++i];
        } else if (!args[i].startsWith('-')) {
            imagePath = args[i];
        }
    }

    if (!imagePath) {
        console.log('Usage: node ocr_reader.js <image> [-l lang] [-o output]');
        process.exit(1);
    }

    if (!fs.existsSync(imagePath)) {
        console.error(`\x1b[31m❌ Файл не найден: ${imagePath}\x1b[0m`);
        process.exit(1);
    }

    console.log('\x1b[36m📖 OCR Reader (JavaScript)\x1b[0m');
    console.log(`📂 Обработка: ${imagePath}`);
    console.log('⏳ Распознавание...');

    const start = Date.now();
    try {
        const result = await Tesseract.recognize(imagePath, lang);
        const text = result.data.text.trim();
        const elapsed = (Date.now() - start) / 1000;

        console.log(`\n\x1b[32mРезультат:\x1b[0m`);
        console.log('─'.repeat(50));
        console.log(text);
        console.log('─'.repeat(50));

        const words = text.split(/\s+/).filter(w => w).length;
        const chars = text.length;
        console.log(`\x1b[36m📊 Статистика:\x1b[0m`);
        console.log(`  Слов: ${words}`);
        console.log(`  Символов: ${chars}`);
        console.log(`  Время: ${elapsed.toFixed(2)} сек`);

        const outFile = outputFile || path.basename(imagePath, path.extname(imagePath)) + '.txt';
        fs.writeFileSync(outFile, text);
        console.log(`\x1b[32m💾 Сохранено в: ${outFile}\x1b[0m`);
    } catch (err) {
        console.error(`\x1b[31m❌ Ошибка: ${err.message}\x1b[0m`);
        process.exit(1);
    }
}

main().catch(console.error);

File: ocr_reader.rs
// ocr_reader.rs — Rust версия

use std::env;
use std::fs;
use std::time::Instant;
use rusty_tesseract::{Image, Args};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let args: Vec<String> = env::args().collect();
    let mut image_path = None;
    let mut lang = "eng".to_string();
    let mut output_file = None;

    let mut i = 1;
    while i < args.len() {
        match args[i].as_str() {
            "-l" | "--lang" => {
                lang = args[i+1].clone();
                i += 2;
            }
            "-o" | "--output" => {
                output_file = Some(args[i+1].clone());
                i += 2;
            }
            _ => {
                if !args[i].starts_with("-") {
                    image_path = Some(args[i].clone());
                }
                i += 1;
            }
        }
    }

    let image_path = image_path.ok_or("Usage: cargo run -- <image> [-l lang] [-o output]")?;

    println!("\x1b[36m📖 OCR Reader (Rust)\x1b[0m");
    println!("📂 Обработка: {}", image_path);
    println!("⏳ Распознавание...");

    let start = Instant::now();
    let img = Image::from_path(&image_path)?;
    let args = Args::default().lang(&lang);
    let text = rusty_tesseract::image_to_string(&img, &args)?;
    let text = text.trim();
    let elapsed = start.elapsed().as_secs_f64();

    println!("\n\x1b[32mРезультат:\x1b[0m");
    println!("{}", "─".repeat(50));
    println!("{}", text);
    println!("{}", "─".repeat(50));

    let words: Vec<&str> = text.split_whitespace().collect();
    println!("\x1b[36m📊 Статистика:\x1b[0m");
    println!("  Слов: {}", words.len());
    println!("  Символов: {}", text.len());
    println!("  Время: {:.2} сек", elapsed);

    let out_file = output_file.unwrap_or_else(|| {
        let path = std::path::Path::new(&image_path);
        path.file_stem().unwrap().to_str().unwrap().to_string() + ".txt"
    });
    fs::write(&out_file, text)?;
    println!("\x1b[32m💾 Сохранено в: {}\x1b[0m", out_file);

    Ok(())
}
